ipapython/fqdn.py

```diff
@@ -0,0 +1,30 @@
+#
+# Copyright (C) 2020  FreeIPA Contributors see COPYING for license
+#
+"""Get host's FQDN
+"""
+import socket
+
+
+def gethostfqdn():
+    hostname = socket.gethostname()
+
+    # optional optimization, consider hostname with dot as FQDN
+    if "." in hostname:
+        return hostname
```

Comment:
Hmm... is this a safe assumption?

Reply:
No, it's actually not safe. hostname may be partly qualified. In that case glibc/NSS can return a FQDN canonical name.